src/Build.cmd

```diff
@@ -2,6 +2,38 @@
 setlocal
 
 SET CMDHOME=%~dp0.
+
+@REM Locate VS 2017 with the proper method
+
+SET VSWHERE_REMOTE_PATH=https://github.com/Microsoft/vswhere/releases/download/1.0.55/vswhere.exe
+SET VSWHERE_LOCAL_PATH=%CMDHOME%\vswhere.exe
+
+if NOT exist "%VSWHERE_LOCAL_PATH%" (
+  echo Downloading vswhere.exe from %VSWHERE_REMOTE_PATH%.
+  powershell -NoProfile -ExecutionPolicy unrestricted -Command "$retryCount = 0; $success = $false; do { try { (New-Object Net.WebClient).DownloadFile('%VSWHERE_REMOTE_PATH%', '%VSWHERE_LOCAL_PATH%'); $success = $true; } catch { if ($retryCount -ge 6) { throw; } else { $retryCount++; Start-Sleep -Seconds (5 * $retryCount); } } } while ($success -eq $false);"
+  if NOT exist "%VSWHERE_LOCAL_PATH%" (
+    echo ERROR: Could not install vswhere correctly.
+    exit /b 1
```

Comment:
I wouldn't exit, I'd try to fallback to using the env variables. In fact, I would first check the variables, and if they aren't there, then you use vswhere. This is faster when the env variables are defined AND allows us to selectively pick the VS tooling we want to use if we don't want VSWHERE to randomly pick one of the installations.
So we don't *depend* on the variables being there, but if they are, we'll honor them